pysr/regressor_sequence.py

```diff
@@ -0,0 +1,189 @@
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to (n_features,).
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+
+        _check_assertions(
+            X,
+            y,
+            Xresampled,
+            self.recursive_history_length,
+            weights,
+            variable_names,
+            X_units,
+            y_units,
+        )
+
+        y = X[self.recursive_history_length :]
+        newX = np.lib.stride_tricks.sliding_window_view(
+            X[:-1].flatten(), self.recursive_history_length * X.shape[1]
+        )[:: y.shape[1], :]
+        y_units = X_units
+        if isinstance(weights, np.ndarray):
+            weights = weights[self.recursive_history_length :]
+        variable_names = self._variable_names(y, variable_names)
+
+        super().fit(
+            newX,
+            y,
+            weights=weights,
+            variable_names=variable_names,
+            X_units=X_units,
+            y_units=y_units,
+            complexity_of_variables=complexity_of_variables,
+        )
+
+        return self
+
+    def predict(self, X, index=None):
+        """
+        Predict y from input X using the equation chosen by `model_selection`.
+
+        You may see what equation is used by printing this object. X should
+        have the same columns as the training data.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Data of shape `(n_times, n_features)`.
+        index : int | list[int]
+            If you want to compute the output of an expression using a
+            particular row of `self.equations_`, you may specify the index here.
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+
+        Returns
+        -------
+        x_predicted : ndarray of shape (n_samples, n_features)
+            Values predicted by substituting `X` into the fitted sequence symbolic
+            regression model.
+
+        Raises
+        ------
+        ValueError
+            Raises if the `best_equation` cannot be evaluated.
+        """
+        X = _check_assertions(X, self.recursive_history_length)[0]
+        X = np.lib.stride_tricks.sliding_window_view(
+            X.flatten(), self.recursive_history_length * np.prod(X.shape[1])
+        )[:: X.shape[1], :]
+        return super().predict(X, index=index)
```

Comment:
I think `predict` should output something that is the same shape as `X`. To do this, I think the predictions should be stacked with a matrix of `NaN`s for rows which do not have enough history to predict them.

Reply:
currently using np.nan and it throws `TypeError: only integer scalar arrays can be converted to a scalar index`
also what would the model output for the rows with NaN's?

Reply:
Not sure what that error is. Maybe concatenating incorrectly or something?

Reply:
The model should automatically output NaNs for those I think.

Reply:
```py
File "/opt/homebrew/lib/python3.12/site-packages/sklearn/utils/validation.py", line 172, in _assert_all_finite_element_wise
    raise ValueError(msg_err)
ValueError: Input contains NaN.
```
maybe not?
not much we can do about that